thorin/util/dl.h

```diff
@@ -7,8 +7,8 @@ namespace thorin::dl {
 
 std::string_view extension(); ///< `".dll"` or `".so"`
 
-void* open(const std::string& filename);
-void* get(void* handle, const std::string& symbol_name);
+void* open(const char* filename);
```

Comment:
can we agree on `string_view`?

Reply:
The problem is that the APIs underneath that expect a `const char*`. So, changing to `std::string_view` means, that we would have to *convert* the original string - which **is** already a `const char*` to `std::string_view` just to *convert* this thing back to `const char*` - which is btw a bit annoying to do:
```c++
void* open(std::string_view sv) {
    std::string str(sv);
    auto file = str.c_str();
    // ...
}
```
This all seemed a bit idiotic to me. Hence, I just used `const char*`. But I do see the consistency argument. So, if you insist, I'll change it to `std::string_view` :)

Reply:
Aah.. Right. That's been the issue, why we're not using string view here ^^ its OK :)